Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/DesiredState.cs

```diff
@@ -21,5 +21,7 @@ public DesiredState(Func<IWebElement, bool> elementStateCondition, string stateN
         public bool IsThrowingNoSuchElementException { get; set; }
 
         public string StateName { get; }
+
+        public string ElementName { get; set; }
```

Comment:
this kinda smells: why should state contain `ElementName`?

Reply:
ok, so containing `IsThrowingNoSuchElementException` property doesn't confuse you? :)
DesiredState is an element-related model class, generally intended to be used internally inside Aquality.Selenium.Core and sibling libraries (such as Aquality.Appium.Mobile etc.). As you might mentioned, it contains not the "State" properties but kind of query parameters for element search, including bool properties like "do we need to throw one exception or another when the search failed". 
Do you have any suggestions how to rework this and not to break backwards compatibility?

Reply:
reworked, removed ElementName property, IElementFinder interface slightly changes